File: Find_the_Index_of_the_First_Occurrence_in_a_String/find_the_index_of_the_first_occurrence_in_a_string.py
from typing import List

class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        """
        Returns the index of the first occurrence of needle in haystack, or -1 if needle is not part of haystack.
        """ 
        return haystack.find(needle)
    
# Example usage:
if __name__ == "__main__":
    solution = Solution()
    haystack = "hello"
    needle = "ll"
    result = solution.strStr(haystack, needle)
    print(f"The index of the first occurrence of '{needle}' in '{haystack}' is: {result}")

File: Find_the_Index_of_the_First_Occurrence_in_a_String/find_the_index_of_the_first_occurrence_in_a_string.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution{
    public:
        int strStr(string haystack, string needle) {
            int haystackLength = haystack.length();
            int needleLength = needle.length();
            
            for (int i = 0; i <= haystackLength - needleLength; i++) {
                int j;
                for (j = 0; j < needleLength; j++) {
                    if (haystack[i + j] != needle[j]) {
                        break;
                    }
                }
                if (j == needleLength) {
                    return i; // Found the first occurrence
                }
            }
            return -1; // Needle not found in haystack
        }
    };

int main() {
    Solution solution;
    string haystack = "hello", needle = "ll";
    int result = solution.strStr(haystack, needle);
    cout << "The index of the first occurrence of needle in haystack is: " << result << endl;
    
    return 0;
}